### ISSUE TITLE
[Bug]: CI case hangs when enable compaction

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#12256]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20211124-62bbc9c
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.0.0rc9.dev16
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: datacoord.enableCompaction: true
          dataCoord.compaction.retentionDuration: 20
```

### Current Behavior

1. Deploy milvus cluster
2. `pytest --tags L0 L1 --host xxxx -n 6`

```
[gw0] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestGetCollectionStats.test_get_collection_stats_after_create_different_index ✓                                  21% ██▏       
[gw1] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestDescribeCollection.test_collection_fields[get_filter_field1-get_vector_field1] ✓                             21% ██▏       
[gw1] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestDescribeCollection.test_describe_collection_after_index_created[get_simple_index1] ✓                         21% ██▏       
[gw1] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestLoadCollection.test_load_collection_after_index[get_simple_index4] ✓                                         21% ██▏       
[gw2] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestLoadCollection.test_load_collection_after_index[get_simple_index0] ✓                                         21% ██▏       
[gw4] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestDescribeCollection.test_describe_collection_after_index_created[get_simple_index7] ✓                         21% ██▏       
[gw1] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestDescribeCollection.test_describe_collection_fields_after_insert[get_filter_field1-get_vector_field0] ✓       21% ██▎       
[gw0] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestGetCollectionStats.test_get_collection_stats_single ✓                                                        21% ██▎       
[gw5] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestLoadCollection.test_release_collection_not_load ✓                                                            21% ██▎       
[gw2] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestDropCollection.test_drop_collection_A ✓                                                                      21% ██▎       
[gw0] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestDescribeCollection.test_describe_collection_fields_after_insert[get_filter_field1-get_vector_field1] ✓       21% ██▎       
[gw1] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestDropCollection.test_drop_collection_not_existed ✓                                                            22% ██▎       
[gw1] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestLoadCollection.test_load_collection_after_index[get_simple_index5] ✓                                         22% ██▎       
[gw3] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestGetCollectionStats.test_get_collection_stats_after_index_created[get_simple_index1] ✓                        22% ██▎       
[gw5] PASSED testcases/test_collection.py 
 testcases/test_collection.py::TestLoadCollection.test_load_collection_after_index[get_simple_index6] ✓                                         22% ██▎     
^C[get_env_variable] failed to get environment variables : 'CI_LOG_PATH', use default path : /tmp/ci_logs/
```
**Ci Logs:**
[ci_logs.tar.gz](https://github.com/milvus-io/milvus/files/7593486/ci_logs.tar.gz)
 

**MIlvus Logs:**
[milvus_logs.tar.gz](https://github.com/milvus-io/milvus/files/7593487/milvus_logs.tar.gz)

**Comment by yanliang567** [comment]
@sunby

**Comment by sunby** [comment]
@congqixia is fixing it.
/assign @congqixia

**Comment by ThreadDao** [comment]
@congqixia  I only know milvus flush hangs, not sure if it is related to compact function.

**Comment by binbinlv** [comment]
@ThreadDao 

please help to verify, thanks.

**Comment by binbinlv** [comment]
Verified, it is not hangs, but milvus is crashed with panic.

**Comment by binbinlv** [comment]
New issue: https://github.com/milvus-io/milvus/issues/12265

**Comment by binbinlv** [comment]
Will re-verify this after issue #12265

**Comment by ThreadDao** [comment]
Fixed `master-20211124-1a8e2e5`
